mbed-client/m2mbase.h

```diff
@@ -481,34 +518,28 @@ protected : // from M2MReportObserver
 
     static bool build_path(StringBuffer<MAX_PATH_SIZE_4> &buffer, const char *s1, uint16_t i1);
 
+    char* stringdup(const char* s);
+
 private:
 
```

Comment:
Nevermind, it needs to be a pointer to allow passing it from the application side also.